core/services/chainlink/application.go

```diff
@@ -560,6 +569,8 @@ func (app *ChainlinkApplication) stop() (err error) {
 
 		if app.Nurse != nil {
 			err = multierr.Append(err, app.Nurse.Close())
+			// when Nurse gets enabled, pyroscope is enabled
+			err = multierr.Append(err, app.prflr.Stop())
```

Comment:
I'm leaning toward having this have it's own pathway and have simpler gating logic than nested in nurse. the logger.StartPyroscope acts on it's own as a gateway for intentional pyroscope setup